### PR TITLE
ui(components) Add mobile layout for StepProgress; Add styleguide exa…

### PR DESCRIPTION
**Pull request by EugeneHlushko**
…mples

- Add mobile layout and necessary styles, variables, elements to render a circle progress bar that can adapt to variable number of steps
- Add values as props to the progress pie so that it can be moved out to a separate file for the ease of maintainability in the future
- Add `forceMobile` option for styleguide development
- Closes https://github.com/opencollective/opencollective/issues/1727

I want to get early feedback and clarify the list of TODO items, so that i can work against a fixed list of todos to complete the task
TODO:
- Because this component will be rendered server-side (SSR), we must render both mobile and desktop views and use CSS for conditional rendering. A nice optimization can be made client-side thanks to the withViewport helper to avoid rendering the view that will not be displayed when viewport is not UNKNOWN
-  Finish styles as per design

### CODE REVIEW

**Comment by EugeneHlushko** [comment]
Screen Shot 2019-10-14 at 9 07 40 PM

File: components/StepsProgress.js [hunk 1]
```diff
@@ -107,43 +207,71 @@ const StepsProgress = ({
...
-              >
-                {getBubbleContent(idx, checked, loading)}
-              </Bubble>
-              <SeparatorLine active={checked} transparent={idx === steps.length - 1} />
+    <StepsOuter className="steps-progress">
+      {forceMobile ? (
+        <StepMobile>
+          <StepsMobileLeft>
+            <p>{steps[mobileStepIdx].name}</p>
+            <p>Next: {mobileNextStepText}</p>
```
Comment (Betree) [inline components/StepsProgress.js]: Needs to be internationalized. See https://docs.opencollective.com/help/developers/translations

File: components/StepsProgress.js [hunk 2]
```diff
@@ -81,6 +81,106 @@ const SeparatorLine = styled(Box)`
...
+  margin-left: auto;
+  width: 48px;
+  height: 48px;
+  align-items: center;
+  justify-content: center;
+  flex-shrink: 0;
+  position: relative;
+`;
+
+const PieProgressWrapper = styled(Box)`
```
Comment (Betree) [inline components/StepsProgress.js]: If we don't use the additional props from components, for example `Box` here (`mx`, `width`...etc), then it's better to use a basic HTML tag - like `styled.div` instead to not add unnecessary classes.

File: components/StepsProgress.js [hunk 3]
```diff
@@ -107,43 +207,71 @@ const StepsProgress = ({
...
+          </StepsMobileLeft>
+          <StepsMobileRight>
+            <PieProgressWrapper>
+              <PieProgress progress={progress} pieSize={pieSize}>
+                <PieHalfCircleLeft progress={progress} pieSize={pieSize} />
+                <PieHalfCircleRight progress={progress} pieSize={pieSize} />
+              </PieProgress>
+              <PieShadow pieSize={pieSize} bgColor={bgColor} />
+            </PieProgressWrapper>
+            {mobileStepIdx + 1} of {steps.length}
```
Comment (Betree) [inline components/StepsProgress.js]: Needs to be internationalized. See https://docs.opencollective.com/help/developers/translations

File: components/StepsProgress.js [hunk 4]
```diff
@@ -81,6 +81,106 @@ const SeparatorLine = styled(Box)`
...
+  ${props => css`
+    border: ${props.pieSize / 10}px solid #3498db;
+    clip: rect(0, ${props.pieSize / 2}px, ${props.pieSize}px, 0);
+  `}
+  border-radius: 50%;
+
+  ${props =>
+    props.progress &&
+    css`
+      border-color: ${props.color};
```
Comment (Betree) [inline components/StepsProgress.js]: Right now we don't need to be able to customize the color, so it's better to use theme's colors.

**Comment by znarf** [comment]
@EugeneHlushko this PR is stale, it will be closed in 7 days if no new activity.

**Pull request closed without merging**